### PR TITLE
chore(pre-commit): add `black` hook

### PR DESCRIPTION
**Pull request by mattyclarkson**
Adds the black formatter hook to the project.

Makes sure that any commits format the Python files.

Ignores the upstream `git-clang-format`.

```
pre-commit install-hooks
pre-commit run --all-files
```

Hoping to get a series of hooks in before contributing functional code changes. Will hopefully align the project on code quality/style.

### CODE REVIEW

**Review by jlebar — APPROVED** [review]

**Comment by mattyclarkson** [comment]
FYI: I don't have write access to merge this, feel free to merge if you are happy. Will continue to upstream PRs. Do you want them as individual small PRs or do you want big PRs with multiple atomic commits?

**Comment by jlebar** [comment]
Oh, right.  Will merge.

> Do you want them as individual small PRs or do you want big PRs with multiple atomic commits?

I'm happy either way.

I actually wrote https://github.com/jlebar/git-pr-chain to allow me to chain PRs (so I could have small PRs but a chain of commits in git).  But I'm told it stopped working, and I'm not currently doing a github-based workflow, so I haven't looked into it.

Thank you for your contributions!

**Pull request merged**
